src/game/interactables_2/091.c

```diff
@@ -0,0 +1,523 @@
+static void Task_807E16C(void);
+static bool16 sub_807E1C4(Sprite_EggUtopia_Launcher *launcher);
+
+void initSprite_EggUtopia_Launcher(MapEntity *me, u16 spriteRegionX, u16 spriteRegionY,
+                                   u8 spriteY, u32 kind)
+{
+    struct Task *t = TaskCreate(Task_807DE98, sizeof(Sprite_EggUtopia_Launcher), 0x2010,
+                                0, TaskDestructor_807DF38);
+    Sprite_EggUtopia_Launcher *launcher = TaskGetStructPtr(t);
+
+    launcher->kind = kind;
+    launcher->posX = TO_WORLD_POS(me->x, spriteRegionX);
+    launcher->posY = TO_WORLD_POS(me->y, spriteRegionY);
+
+    launcher->unk4C = me->d.sData[0] * TILE_WIDTH;
+    launcher->unk4E = me->d.sData[1] * TILE_WIDTH;
+    launcher->unk50 = launcher->unk4C + me->d.uData[2] * TILE_WIDTH;
+    launcher->unk52 = launcher->unk4E + me->d.uData[3] * TILE_WIDTH;
+
+    launcher->base.regionX = spriteRegionX;
+    launcher->base.regionY = spriteRegionY;
+    launcher->base.me = me;
+    launcher->base.spriteX = me->x;
+    launcher->base.spriteY = spriteY;
+
+    switch (launcher->kind) {
+        case LAUNCHER_KIND(LAUN_DIR_LEFT, LAUN_GRAVITY_DOWN): {
+            launcher->unk54 = Q_24_8(launcher->posX + launcher->unk50);
+            launcher->unk58 = Q_24_8(launcher->posY + launcher->unk52);
+        } break;
+
+        case LAUNCHER_KIND(LAUN_DIR_RIGHT, LAUN_GRAVITY_DOWN): {
+            launcher->unk54 = Q_24_8(launcher->posX + launcher->unk4C);
+            launcher->unk58 = Q_24_8(launcher->posY + launcher->unk52);
+        } break;
+
+        case LAUNCHER_KIND(LAUN_DIR_LEFT, LAUN_GRAVITY_UP): {
+            launcher->unk54 = Q_24_8(launcher->posX + launcher->unk50);
+            launcher->unk58 = Q_24_8(launcher->posY + launcher->unk4E);
+        } break;
+
+        case LAUNCHER_KIND(LAUN_DIR_RIGHT, LAUN_GRAVITY_DOWN): {
+            launcher->unk54 = Q_24_8(launcher->posX + launcher->unk4C);
+            launcher->unk58 = Q_24_8(launcher->posY + launcher->unk4E);
+        } break;
+    }
+
+    {
+        s32 *xs, *ys;
+        int i = 0;
+#ifndef NON_MATCHING
+        register void *s2 asm("r4") = &launcher->s;
+#endif
+        for (; i < ARRAY_COUNT(launcher->unk60); i++) {
+            launcher->unk60[i][0] = launcher->unk54;
+            launcher->unk60[i][1] = launcher->unk58;
+        }
+
+        {
+#ifndef NON_MATCHING
+            register Sprite *s asm("r5") = s2;
+#else
+            Sprite *s = &launcher->s;
+#endif
+            s->unk1A = 0x180;
+            s->graphics.size = 0;
+            s->unk14 = 0;
+            s->unk1C = 0;
+
+            launcher->s.unk21 = 0xFF;
+            launcher->s.unk22 = 0x10;
+            launcher->s.focused = 0;
+
+            s->unk28->unk0 = -1;
+            s->unk10 = SPRITE_FLAG_PRIORITY(2);
+            s->graphics.dest = VramMalloc(EGG_UTO_LAUNCHER_TILE_COUNT);
+            s->graphics.anim = SA2_ANIM_LAUNCHER_EGG_UTO;
+            launcher->s.variant = 0;
+
+#ifndef NON_MATCHING
```

Comment:
Not sure this counts as a fake match? Typically nonmatching is used for that?

Reply:
Well, it is a fake-match, but I didn't know how else to make it work.

Reply:
Is it fake? It's what they wrote

Reply:
Maybe it shows there is some macro which made them non aware of it being redundant

Reply:
OH, I thought you were talking about the register vars.
Github doesn't show the correct lines in the preview so I didn't realize what you were talking about.

No, I don't think that is a fake-match, but yes it's likely due to them using a macro.